### PR TITLE
Fix tunnel monitor check

### PR DESCRIPTION
**Pull request by samsymons**
Please review the release process for BrowserServicesKit [here](https://app.asana.com/0/1200194497630846/1200837094583426).

**Required**:

Task/Issue URL: https://app.asana.com/0/414235014887631/1206397132422633/f
iOS PR: https://github.com/duckduckgo/iOS/pull/2361
macOS PR: https://github.com/duckduckgo/macos-browser/pull/2096
What kind of version bump will this require?: Patch

**Description**:

This PR improves the tunnel monitor check by ignoring handshake values that are equal to zero, avoiding false failure pings.

**Steps to test this PR**:
1.
1.

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16
* [ ] macOS 10.15
* [ ] macOS 11
* [ ] macOS 12

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Sources/NetworkProtection/PacketTunnelProvider.swift [hunk 1]
```diff
@@ -994,9 +994,11 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
             connectionStatus = .connected(connectedDate: Date())
         }
 
-        guard !isKeyExpired else {
-            await rekey()
-            return
+        if !settings.disableRekeying {
```
Comment (diegoreymendez) [inline Sources/NetworkProtection/PacketTunnelProvider.swift:997]: Please bear with me for a sec as I'm confused by this.  It certainly looks right to do this if there's a setting, but I'm lacking context about why this was added.

My question is two-fold and it's mostly to catch up with our plans:
1. Why was this added to begin with?
2. What's the plan for rekeying that makes us want to disable it optionally?

**Review by diegoreymendez — APPROVED** [review]
Works well for me.

File: Sources/NetworkProtection/PacketTunnelProvider.swift (hunk 1 above)
Comment (samsymons) [reply Sources/NetworkProtection/PacketTunnelProvider.swift:997]: This was a debug option setting that I added last year when I was talking with Aitor about the effects of removing rekeying - it's not used for prod users in any way. There's a discussion task in Asana hashing out the merits of removing rekeying, but I forget where it got to.

This change here is just a bugfix that tripped me up while testing this feature - I had the setting to disable rekeying turned on, which was causing the app to always get to this `isKeyExpired` check and never start the tunnel monitor.
